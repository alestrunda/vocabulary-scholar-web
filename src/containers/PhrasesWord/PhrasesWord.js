import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import PhraseWordPreview from '../../components/PhraseWordPreview';
import Popup from '../../components/Popup';
import PhraseWordSettings from '../../components/PhraseWordSettings';
import RaisedButton from 'material-ui/RaisedButton';
import Dialog from 'material-ui/Dialog';
import Snackbar from 'material-ui/Snackbar';
import { appSetSelectedPhrase } from '../../actions/app';
import {
  wordAddPhrase,
  wordEditPhrase,
  wordRemovePhrase,
} from '../../actions/word';
import { SNACKBAR_HIDE_DURATION } from '../../constants';

export class PhrasesWord extends React.Component {
  state = {
    isPopupNewPhraseOpened: false,
    isPopupEditPhraseOpened: false,
    isDialogPhraseRemoveOpened: false,
    isSnackBarPhraseAddedOpened: false,
    isSnackBarPhraseEditedOpened: false,
    isSnackBarPhraseRemovedOpened: false,
  };

  componentDidMount() {
    if (this.props.phraseSelected) {
      //open popup with phrase settings right away if perticular phrase set
      this.setState({
        isPopupNewPhraseOpened: true,
      });
    }
  }

  componentDidUpdate(prevProps) {
    if (
      prevProps.phraseSelected !== this.props.phraseSelected &&
      !this.state.isPopupNewPhraseOpened
    ) {
      //open popup with phrase settings right away if perticular phrase set
      this.setState({
        isPopupNewPhraseOpened: true,
      });
    }
  }

  componentWillUnmount() {
    //selected phrase is saved in store, so it should be cleaned when the work with phrases is done,
    //we want to see empty form next time we open it
    this.props.onPhraseSelected('');
  }

  handleNewPhraseBtnClick = () => {
    this.setState({
      isPopupNewPhraseOpened: true,
    });
  };

  closeNewPhrasePopup = () => {
    this.setState({
      isPopupNewPhraseOpened: false,
    });
  };

  closeEditPhrasePopup = () => {
    this.setState({
      isPopupEditPhraseOpened: false,
    });
  };

  handleDialogPhraseRemoveConfirm = () => {
    this.props.onWordRemovePhrase(
      this.props.wordID,
      this.state.phraseIDToRemove
    );
    this.setState({
      isSnackBarPhraseRemovedOpened: true,
    });
  };

  handleDialogPhraseRemoveClose = () => {
    this.setState({
      isDialogPhraseRemoveOpened: false,
    });
  };

  handlePhraseRemove = phrase => {
    this.setState({
      isDialogPhraseRemoveOpened: true,
      phraseIDToRemove: phrase,
    });
  };

  handlePhraseEdit = phraseID => {
    const phraseToEdit = this.props.phrases.find(
      phrase => phrase.id === phraseID
    );
    this.setState({
      isPopupEditPhraseOpened: true,
      phraseToEdit,
    });
  };

  editPhrase = (phrase, startIndex, endIndex) => {
    this.props.onEditPhrase(
      this.props.wordID,
      this.state.phraseToEdit.id,
      phrase,
      startIndex,
      endIndex
    );
    this.closeEditPhrasePopup();
    this.setState({
      isSnackBarPhraseEditedOpened: true,
    });
  };

  savePhrase = (phrase, startIndex, endIndex) => {
    this.props.onAddPhrase(this.props.wordID, phrase, startIndex, endIndex);
    this.closeNewPhrasePopup();
    this.setState({
      isSnackBarPhraseAddedOpened: true,
    });
  };

  handleSnackBarPhraseAddedClose = () => {
    this.setState({
      isSnackBarPhraseAddedOpened: false,
    });
  };

  handleSnackBarPhraseEditedClose = () => {
    this.setState({
      isSnackBarPhraseEditedOpened: false,
    });
  };

  handleSnackBarPhraseRemovedClose = () => {
    this.setState({
      isSnackBarPhraseRemovedOpened: false,
    });
  };

  renderPhraseRemoveDialog() {
    const phraseToRemove = this.props.phrases.find(
      phrase => phrase.id === this.state.phraseIDToRemove
    );
    if (phraseToRemove === undefined) return; //phrase not found
    const dialogActions = [
      <RaisedButton
        className="ml10"
        key="btn-cancel"
        label="Cancel"
        onClick={this.handleDialogPhraseRemoveClose}
      />,
      <RaisedButton
        className="ml10"
        key="btn-remove"
        label="Remove"
        secondary={true}
        onClick={this.handleDialogPhraseRemoveConfirm}
      />,
    ];
    return (
      <Dialog
        actions={dialogActions}
        modal={false}
        open={this.state.isDialogPhraseRemoveOpened}
        onRequestClose={this.handleDialogPhraseRemoveClose}
      >
        <p className="mb10 text-weight-bold text-big">
          Do you wish to remove this phrase?
        </p>
        <p className="phrase-excerpt">&quot;{phraseToRemove.phrase}&quot;</p>
      </Dialog>
    );
  }

  render() {
    return (
      <div className={this.props.className} style={this.props.style}>
        {this.props.phrases.length === 0 && <p>None</p>}
        <div className="mb20">
          {this.props.phrases.map(phrase => (
            <PhraseWordPreview
              key={phrase.id}
              onEdit={this.handlePhraseEdit}
              onRemove={this.handlePhraseRemove}
              {...phrase}
            />
          ))}
        </div>
        <div className="text-center">
          <RaisedButton
            primary
            label="New phrase"
            onClick={this.handleNewPhraseBtnClick}
            data-test-id="btn-new-phrase"
          />
        </div>
        <Popup
          isActive={this.state.isPopupNewPhraseOpened}
          onClose={this.closeNewPhrasePopup}
          title="New phrase"
          data-test-id="popup-new-phrase"
        >
          <div className="text-center">
            <PhraseWordSettings
              phrase={{ phrase: this.props.phraseSelected }}
              word={this.props.word}
              wordID={this.props.wordID}
              onSubmit={this.savePhrase}
            />
          </div>
        </Popup>
        <Popup
          isActive={this.state.isPopupEditPhraseOpened}
          onClose={this.closeEditPhrasePopup}
          title="Edit phrase"
        >
          <div className="text-center">
            <PhraseWordSettings
              phrase={this.state.phraseToEdit}
              word={this.props.word}
              wordID={this.props.wordID}
              onSubmit={this.editPhrase}
              submitText="Edit phrase"
            />
          </div>
        </Popup>
        <Snackbar
          open={this.state.isSnackBarPhraseAddedOpened}
          message="New phrase added"
          autoHideDuration={SNACKBAR_HIDE_DURATION}
          onRequestClose={this.handleSnackBarPhraseAddedClose}
        />
        <Snackbar
          open={this.state.isSnackBarPhraseEditedOpened}
          message="Selected phrase was changed"
          autoHideDuration={SNACKBAR_HIDE_DURATION}
          onRequestClose={this.handleSnackBarPhraseEditedClose}
        />
        <Snackbar
          open={this.state.isSnackBarPhraseRemovedOpened}
          message="Selected phrase was removed"
          autoHideDuration={SNACKBAR_HIDE_DURATION}
          onRequestClose={this.handleSnackBarPhraseRemovedClose}
        />
        {this.renderPhraseRemoveDialog()}
      </div>
    );
  }
}

PhrasesWord.propTypes = {
  wordID: PropTypes.string.isRequired,
  word: PropTypes.string.isRequired,
  phrases: PropTypes.array,
  className: PropTypes.string,
  phraseSelected: PropTypes.string,
  style: PropTypes.object,
  onWordRemovePhrase: PropTypes.func.isRequired,
  onAddPhrase: PropTypes.func.isRequired,
  onEditPhrase: PropTypes.func.isRequired,
  onPhraseSelected: PropTypes.func.isRequired,
};

PhrasesWord.defaultProps = {
  phrases: [],
};

const mapStateToProps = state => ({
  phraseSelected: state.app.phraseSelected,
});

const mapDispatchToProps = dispatch => ({
  onWordRemovePhrase: (wordID, phraseID) =>
    dispatch(wordRemovePhrase(wordID, phraseID)),
  onAddPhrase: (wordID, phrase, startIndex, endIndex) =>
    dispatch(wordAddPhrase({ wordID, phrase, startIndex, endIndex })),
  onEditPhrase: (wordID, phraseID, phrase, startIndex, endIndex) =>
    dispatch(
      wordEditPhrase({ wordID, phraseID, phrase, startIndex, endIndex })
    ),
  onPhraseSelected: phrase => dispatch(appSetSelectedPhrase(phrase)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(PhrasesWord);
