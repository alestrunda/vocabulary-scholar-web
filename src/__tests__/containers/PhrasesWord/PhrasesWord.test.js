import React from 'react';
import { shallow } from 'enzyme';

import { PhrasesWord } from '../../../containers/PhrasesWord/PhrasesWord';

const onWordRemovePhrase = jest.fn(),
  onAddPhrase = jest.fn(),
  onEditPhrase = jest.fn(),
  onPhraseSelected = jest.fn();

it('renders correctly', () => {
  const root = shallow(
    <PhrasesWord
      word="car"
      wordID="car"
      phraseSelected="Lorem ipsum"
      onWordRemovePhrase={onWordRemovePhrase}
      onAddPhrase={onAddPhrase}
      onEditPhrase={onEditPhrase}
      onPhraseSelected={onPhraseSelected}
    />
  );
  expect(root).toMatchSnapshot();
});

it('popup new phrase opened when phrase selected', () => {
  const root = shallow(
    <PhrasesWord
      word="car"
      wordID="car"
      onWordRemovePhrase={onWordRemovePhrase}
      phraseSelected="Lorem ipsum"
      onAddPhrase={onAddPhrase}
      onEditPhrase={onEditPhrase}
      onPhraseSelected={onPhraseSelected}
    />
  );
  expect(root.find('[data-test-id="popup-new-phrase"]').props().isActive).toBe(
    true
  );
});

it('opens new phrase popup', () => {
  const root = shallow(
    <PhrasesWord
      word="car"
      wordID="car"
      onWordRemovePhrase={onWordRemovePhrase}
      phraseSelected=""
      onAddPhrase={onAddPhrase}
      onEditPhrase={onEditPhrase}
      onPhraseSelected={onPhraseSelected}
    />
  );
  expect(root.find('[data-test-id="popup-new-phrase"]').props().isActive).toBe(
    false
  );
  root.find('[data-test-id="btn-new-phrase"]').simulate('click');
  expect(root.find('[data-test-id="popup-new-phrase"]').props().isActive).toBe(
    true
  );
});
