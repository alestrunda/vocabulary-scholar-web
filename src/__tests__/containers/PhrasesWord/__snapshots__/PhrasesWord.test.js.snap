// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PhrasesWord
    onAddPhrase={[MockFunction]}
    onEditPhrase={[MockFunction]}
    onPhraseSelected={[MockFunction]}
    onWordRemovePhrase={[MockFunction]}
    phraseSelected="Lorem ipsum"
    phrases={Array []}
    word="car"
    wordID="car"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          None
        </p>,
        <div
          className="mb20"
        />,
        <div
          className="text-center"
        >
          <RaisedButton
            data-test-id="btn-new-phrase"
            disabled={false}
            fullWidth={false}
            label="New phrase"
            labelPosition="after"
            onClick={[Function]}
            primary={true}
            secondary={false}
          />
        </div>,
        <Popup
          data-test-id="popup-new-phrase"
          isActive={true}
          onClose={[Function]}
          title="New phrase"
        >
          <div
            className="text-center"
          >
            <PhraseWordSettings
              onSubmit={[Function]}
              phrase={
                Object {
                  "phrase": "Lorem ipsum",
                }
              }
              submitText="Save phrase"
              word="car"
              wordID="car"
            />
          </div>
        </Popup>,
        <Popup
          isActive={false}
          onClose={[Function]}
          title="Edit phrase"
        >
          <div
            className="text-center"
          >
            <PhraseWordSettings
              onSubmit={[Function]}
              submitText="Edit phrase"
              word="car"
              wordID="car"
            />
          </div>
        </Popup>,
        <Snackbar
          autoHideDuration={4000}
          message="New phrase added"
          onRequestClose={[Function]}
          open={false}
        />,
        <Snackbar
          autoHideDuration={4000}
          message="Selected phrase was changed"
          onRequestClose={[Function]}
          open={false}
        />,
        <Snackbar
          autoHideDuration={4000}
          message="Selected phrase was removed"
          onRequestClose={[Function]}
          open={false}
        />,
        undefined,
      ],
      "className": undefined,
      "style": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "None",
        },
        "ref": null,
        "rendered": "None",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "mb20",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <RaisedButton
            data-test-id="btn-new-phrase"
            disabled={false}
            fullWidth={false}
            label="New phrase"
            labelPosition="after"
            onClick={[Function]}
            primary={true}
            secondary={false}
          />,
          "className": "text-center",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data-test-id": "btn-new-phrase",
            "disabled": false,
            "fullWidth": false,
            "label": "New phrase",
            "labelPosition": "after",
            "onClick": [Function],
            "primary": true,
            "secondary": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <div
            className="text-center"
          >
            <PhraseWordSettings
              onSubmit={[Function]}
              phrase={
                Object {
                  "phrase": "Lorem ipsum",
                }
              }
              submitText="Save phrase"
              word="car"
              wordID="car"
            />
          </div>,
          "data-test-id": "popup-new-phrase",
          "isActive": true,
          "onClose": [Function],
          "title": "New phrase",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <PhraseWordSettings
              onSubmit={[Function]}
              phrase={
                Object {
                  "phrase": "Lorem ipsum",
                }
              }
              submitText="Save phrase"
              word="car"
              wordID="car"
            />,
            "className": "text-center",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onSubmit": [Function],
              "phrase": Object {
                "phrase": "Lorem ipsum",
              },
              "submitText": "Save phrase",
              "word": "car",
              "wordID": "car",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <div
            className="text-center"
          >
            <PhraseWordSettings
              onSubmit={[Function]}
              submitText="Edit phrase"
              word="car"
              wordID="car"
            />
          </div>,
          "isActive": false,
          "onClose": [Function],
          "title": "Edit phrase",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <PhraseWordSettings
              onSubmit={[Function]}
              submitText="Edit phrase"
              word="car"
              wordID="car"
            />,
            "className": "text-center",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onSubmit": [Function],
              "phrase": undefined,
              "submitText": "Edit phrase",
              "word": "car",
              "wordID": "car",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoHideDuration": 4000,
          "message": "New phrase added",
          "onRequestClose": [Function],
          "open": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoHideDuration": 4000,
          "message": "Selected phrase was changed",
          "onRequestClose": [Function],
          "open": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoHideDuration": 4000,
          "message": "Selected phrase was removed",
          "onRequestClose": [Function],
          "open": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            None
          </p>,
          <div
            className="mb20"
          />,
          <div
            className="text-center"
          >
            <RaisedButton
              data-test-id="btn-new-phrase"
              disabled={false}
              fullWidth={false}
              label="New phrase"
              labelPosition="after"
              onClick={[Function]}
              primary={true}
              secondary={false}
            />
          </div>,
          <Popup
            data-test-id="popup-new-phrase"
            isActive={true}
            onClose={[Function]}
            title="New phrase"
          >
            <div
              className="text-center"
            >
              <PhraseWordSettings
                onSubmit={[Function]}
                phrase={
                  Object {
                    "phrase": "Lorem ipsum",
                  }
                }
                submitText="Save phrase"
                word="car"
                wordID="car"
              />
            </div>
          </Popup>,
          <Popup
            isActive={false}
            onClose={[Function]}
            title="Edit phrase"
          >
            <div
              className="text-center"
            >
              <PhraseWordSettings
                onSubmit={[Function]}
                submitText="Edit phrase"
                word="car"
                wordID="car"
              />
            </div>
          </Popup>,
          <Snackbar
            autoHideDuration={4000}
            message="New phrase added"
            onRequestClose={[Function]}
            open={false}
          />,
          <Snackbar
            autoHideDuration={4000}
            message="Selected phrase was changed"
            onRequestClose={[Function]}
            open={false}
          />,
          <Snackbar
            autoHideDuration={4000}
            message="Selected phrase was removed"
            onRequestClose={[Function]}
            open={false}
          />,
          undefined,
        ],
        "className": undefined,
        "style": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "None",
          },
          "ref": null,
          "rendered": "None",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "mb20",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <RaisedButton
              data-test-id="btn-new-phrase"
              disabled={false}
              fullWidth={false}
              label="New phrase"
              labelPosition="after"
              onClick={[Function]}
              primary={true}
              secondary={false}
            />,
            "className": "text-center",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "data-test-id": "btn-new-phrase",
              "disabled": false,
              "fullWidth": false,
              "label": "New phrase",
              "labelPosition": "after",
              "onClick": [Function],
              "primary": true,
              "secondary": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <div
              className="text-center"
            >
              <PhraseWordSettings
                onSubmit={[Function]}
                phrase={
                  Object {
                    "phrase": "Lorem ipsum",
                  }
                }
                submitText="Save phrase"
                word="car"
                wordID="car"
              />
            </div>,
            "data-test-id": "popup-new-phrase",
            "isActive": true,
            "onClose": [Function],
            "title": "New phrase",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <PhraseWordSettings
                onSubmit={[Function]}
                phrase={
                  Object {
                    "phrase": "Lorem ipsum",
                  }
                }
                submitText="Save phrase"
                word="car"
                wordID="car"
              />,
              "className": "text-center",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onSubmit": [Function],
                "phrase": Object {
                  "phrase": "Lorem ipsum",
                },
                "submitText": "Save phrase",
                "word": "car",
                "wordID": "car",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <div
              className="text-center"
            >
              <PhraseWordSettings
                onSubmit={[Function]}
                submitText="Edit phrase"
                word="car"
                wordID="car"
              />
            </div>,
            "isActive": false,
            "onClose": [Function],
            "title": "Edit phrase",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <PhraseWordSettings
                onSubmit={[Function]}
                submitText="Edit phrase"
                word="car"
                wordID="car"
              />,
              "className": "text-center",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onSubmit": [Function],
                "phrase": undefined,
                "submitText": "Edit phrase",
                "word": "car",
                "wordID": "car",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoHideDuration": 4000,
            "message": "New phrase added",
            "onRequestClose": [Function],
            "open": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoHideDuration": 4000,
            "message": "Selected phrase was changed",
            "onRequestClose": [Function],
            "open": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoHideDuration": 4000,
            "message": "Selected phrase was removed",
            "onRequestClose": [Function],
            "open": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
